packages/kokkos-kernels/src/common/KokkosKernels_Utils.hpp

```diff
@@ -1916,8 +1900,56 @@ void init_view_withscalar(typename in_row_view_t::size_type num_elements, in_row
   MyExecSpace().fence();
 }
 
+//A sum-reduction scalar representing a fixed-size array.
```

Comment:
This feels like a thing that belongs in Kokkos.  It looks related to `Kokkos::array`.

Reply:
@mhoemmen It's similar in that it's a fixed-size array, but Kokkos::Array doesn't have operator+ built in which I need to define the reduction. I could replace the ``data`` member with a Kokkos::Array but I would still have to keep ``array_sum_reduce``.

Reply:
There is an opportunity to remove duplicate code though, since @lucbv made exactly the same thing for SPMV.

Reply:
We should consider it at least as part of Kokkos Kernels Utils, if it doesn't belong in Kokkos. Could be a different commit though.

Reply:
> It's similar in that it's a fixed-size array, but Kokkos::Array doesn't have operator+ built in which I need to define the reduction.

I was just thinking we could use Kokkos::Array if it exists, but no worries :)